Bucketeer/Sources/Internal/Evaluation/EvaluationSQLDao.swift

```diff
@@ -16,11 +18,13 @@ final class EvaluationDaoImpl: EvaluationDao {
         try db.select(EvaluationEntity(), conditions: [.equal(column: "userId", value: userId)])
     }
 
-    func deleteAllAndInsert(userId: String, evaluations: [Evaluation]) throws {
-        let entities = try evaluations.map {
-            try EvaluationEntity(model: $0)
-        }
+    func deleteAll(userId: String) throws {
         try db.delete(EvaluationEntity(), condition: .equal(column: "userId", value: userId))
-        try db.insert(entities)
```

Comment:
The intention of this `deleteAllAndInsert` is to make it atomic(though it seems iOS SDK doesn't use transaction).

So could you keep it and wrap it with a transaction?

Reply:
Let me check on that point.
The previous implementation didn't use transaction

Reply:
I added a transaction logic to the SQLite database class  
https://github.com/bucketeer-io/ios-client-sdk/pull/19/files#diff-dbcfd87b7762ef66ebdf00a4be7ffdc645503b6be816f5d9497db8dca5a045ceR126

I exposed the transaction in `EvaluationDao`
https://github.com/bucketeer-io/ios-client-sdk/pull/19/files#diff-b6582b5d8cdfedd3ae9fdc27856a16f31cfa5c0f6e177447ac2826d58bcc5508R31

and using it on `EvalutionStorageImpl`
https://github.com/bucketeer-io/ios-client-sdk/pull/19/files#diff-7f5ac49b63ea2e6003b8e369a82a2d0acde26aed9df8e3694e158502ddb71b9dR77